main/src/ca/uwaterloo/flix/language/errors/RedundancyError.scala

```diff
@@ -341,6 +341,30 @@ object RedundancyError {
     })
   }
 
+  case class DiscardedValue(loc: SourceLocation) extends RedundancyError {
+    def summary: String = "Non-unit expression value is implicitly discarded."
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> Implicitly discarded value: The non-unit expression value is not used.
```

Comment:
Maybe this message is confusing because it talks about "discarded" but then the solution is to use discard!

How about something like: Unused non-unit value of type: '...'? or similar?